lib/assert.js

```diff
@@ -958,5 +959,100 @@ export class Assertions {
 			pass();
 			return true;
 		});
+
+		this.unorderedEqual = withSkip((actual, expected, message) => {
+			if (!checkMessage('unorderedEqual', message)) {
+				return false;
+			}
+
+			const actualInfo = checkValueForUnorderedEqual(actual);
+
+			if (!actualInfo.isValid) {
+				fail(new AssertionError({
+					assertion: 'unorderedEqual',
+					improperUsage: true,
+					message: '`t.unorderedEqual` only compares Maps, Sets, and arrays',
+					values: [formatWithLabel('Called with:', actual)],
+				}));
+				return false;
+			}
+
+			const expectedInfo = checkValueForUnorderedEqual(expected);
+
+			if (!expectedInfo.isValid) {
+				fail(new AssertionError({
+					assertion: 'unorderedEqual',
+					improperUsage: true,
+					message: '`t.unorderedEqual` only compares Maps, Sets, and arrays',
+					values: [formatWithLabel('Called with:', expected)],
+				}));
+				return false;
+			}
+
+			if (actualInfo.size !== expectedInfo.size) {
+				fail(new AssertionError({
+					assertion: 'unorderedEqual',
+					message: 'size must be equal',
```

Comment:
I think this should be the `message` argument. You can then do something like this:

https://github.com/avajs/ava/blob/a39dab098154c8d4fbfd39a85a72e13e71004c7e/lib/assert.js#L332-L333